test/parallel_api/experimental/transform_async_unary_zip.pass.cpp

```diff
@@ -0,0 +1,124 @@
+// -*- C++ -*-
+//===-- transform_unary.pass.cpp ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// Tests for transform
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#    include "oneapi/dpl/async"
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#include "support/utils.h"
+
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <typename InputIterator, typename OutputIterator>
+void
+check_and_reset(InputIterator first, InputIterator last, OutputIterator out_first)
+{
+    typedef typename ::std::iterator_traits<OutputIterator>::value_type Out;
+    typename ::std::iterator_traits<OutputIterator>::difference_type k = 0;
+    for (; first != last; ++first, ++out_first, ++k)
+    {
+        // check
+        Out expected = 1 - std::get<0>(*first) - std::get<1>(*first);
+        Out actual = *out_first;
+        EXPECT_EQ(expected, actual, "wrong value in output sequence");
+        // reset
+        *out_first = k % 7 != 4 ? 7 * k - 5 : 0;
+    }
+}
+
+template <typename T1, typename T2>
+struct test_one_policy
+{
+    template <typename Policy, typename InputIterator, typename OutputIterator, typename UnaryOp>
+    void
+    operator()(Policy&& exec, InputIterator first, InputIterator last, OutputIterator out_first,
+               OutputIterator out_last, UnaryOp op)
+    {
+        auto result = oneapi::dpl::experimental::transform_async(exec, first, last, out_first, op);
```

Comment:
Do I understand correctly that we have just one difference with call oneapi::dpl::experimental::transform_async instead of ::std::transform (dpl::transform) comparing with file `transform_unary_zip.pass.cpp` ?

Reply:
1) As far a I see we use predicate with std::tuple on the both files: `transform_unary_zip.pass.cpp` and `transform_async_unary_zip.pass.cpp`...
2) Please read my opinion above (about the code duplication).

Reply:
Could you please prepare some PR with fix for that?

Reply:
@SergeyKopienko, you may review the new commits within this PR.